indico/modules/events/persons/client/js/Email.jsx

```diff
@@ -0,0 +1,229 @@
+      return handleSubmitError(err);
+    }
+  };
+
+  const renderForm = ({handleSubmit, submitting, submitSucceeded, values, hasValidationErrors}) => (
+    <Form loading={loading} onSubmit={handleSubmit}>
+      {submitSucceeded && (
+        <Message positive>
+          <Translate as={Message.Header}>Your email has been sent.</Translate>
+          <Translate as="p">
+            <Param
+              name="count"
+              value={Object.values(recipientData).reduce((acc, v) => acc + v.length, 0)}
+            />{' '}
+            emails have been sent.
+          </Translate>
+        </Message>
+      )}
+      <Form.Field>
+        <Translate as="label">From</Translate>
+        <FinalDropdown
+          name="fromAddress"
+          scrolling
+          selection
+          options={senders.map(([value, text]) => ({value, text}))}
+        />
+      </Form.Field>
+      <Form.Field>
+        <Translate as="label">Subject</Translate>
+        <FinalField name="subject">
+          {({input: {value, ...rest}}) => (
+            <Input value={preview ? preview.subject : value} disabled={!!preview} {...rest} />
+          )}
+        </FinalField>
+      </Form.Field>
+      <Form.Field>
+        <Translate as="label">Email body</Translate>
+        <FinalField name="body" validate={validators.required}>
+          {({input, meta}) => (
+            <>
+              {meta.touched && (meta.error || meta.submitError) && (
+                <Label basic color="red" pointing="below">
+                  {meta.error || meta.submitError}
+                </Label>
+              )}
+              <TextEditor
+                {...input}
+                value={preview ? preview.body : input.value}
+                onChange={(_, editor) => !disableChange.current && input.onChange(editor.getData())}
+                disabled={!!preview}
+                required
+              />
+            </>
+          )}
+        </FinalField>
+      </Form.Field>
+      <Form.Field>
+        <Translate as="label">Recipients</Translate>
+        <TextArea rows={3} value={recipients.join(', ')} readOnly />
+      </Form.Field>
+      <Form.Field>
+        <Translate as="label">Send copy to me</Translate>
+        <FinalCheckbox name="copyForSender" label={Translate.string('Send copy to me')} toggle />
+        <Translate as="p" className="field-description">
+          Send copy of each email to my mailbox
+        </Translate>
+      </Form.Field>
+      <Translate
+        as={Button}
+        type="submit"
+        disabled={submitting || submitSucceeded || hasValidationErrors}
+        primary
+      >
+        Send
+      </Translate>
+      <Button type="button" active={!!preview} onClick={() => togglePreview(values)}>
+        {!preview ? <Translate>Preview</Translate> : <Translate>Edit</Translate>}
+      </Button>
+    </Form>
+  );
+
+  return (
+    <FinalForm
+      initialValues={{
+        fromAddress: senders[0] && senders[0][0],
```

Comment:
I think you always have at least one entry (yourself) available here so the `senders[0] && ` may not be needed

Reply:
We don't yet before the data loads, and I'm defaulting to empty list `[]`.

Reply:
I was defaulting it to an empty list `[]` before it got loaded. 
Anyway, this needs a bit of refactoring between state and form data.